WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -0,0 +1,168 @@
+using NBitcoin;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.TreeDataGrid;
+using WalletWasabi.Fluent.ViewModels.CoinControl.Core;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Cells;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Headers;
+
+namespace WalletWasabi.Fluent.ViewModels.CoinControl;
+
+[NavigationMetaData(
+	Title = "Coin Selection",
+	Caption = "",
+	IconName = "wallet_action_send",
+	NavBarPosition = NavBarPosition.None,
+	Searchable = false,
+	NavigationTarget = NavigationTarget.DialogScreen)]
+public partial class SelectCoinsDialogViewModel : DialogViewModelBase<IEnumerable<SmartCoin>>
+{
+	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel)
+	{
+		var pockets = walletViewModel.Wallet.GetPockets();
+		var items = CreateItems(pockets);
+
+		var pocketColumn = PocketColumn();
+		Source = new HierarchicalTreeDataGridSource<CoinControlItemViewModelBase>(items)
+		{
+			Columns =
+			{
+				ChildrenColumn(),
+				IndicatorsColumn(),
+				AmountColumn(),
+				PrivacyScore(),
+				pocketColumn
+			}
+		};
+		Source.SortBy(pocketColumn, ListSortDirection.Descending);
+		Source.RowSelection!.SingleSelect = true;
+		Source.RowSelection!.SingleSelect = true;
+
+		SetupCancel(false, true, false);
+		EnableBack = true;
+	}
+
+	public HierarchicalTreeDataGridSource<CoinControlItemViewModelBase> Source { get; }
+
+	private static IColumn<CoinControlItemViewModelBase> ChildrenColumn()
+	{
+		return new HierarchicalExpanderColumn<CoinControlItemViewModelBase>(
+			new PrivacyTextColumn<CoinControlItemViewModelBase>("", _ => "", GridLength.Auto, null),
+			group => group.Children,
+			node => node.Children.Count > 1,
+			node => node.IsExpanded);
+	}
+
+	private static IColumn<CoinControlItemViewModelBase> AmountColumn()
+	{
+		return new PrivacyTextColumn<CoinControlItemViewModelBase>(
+			"Amount",
+			node => node.Amount.ToFormattedString(),
+			GridLength.Auto,
+			new ColumnOptions<CoinControlItemViewModelBase>
+			{
+				CompareAscending = SortAscending<CoinControlItemViewModelBase, Money>(x => x.Amount),
+				CompareDescending = SortDescending<CoinControlItemViewModelBase, Money>(x => x.Amount)
+			});
+	}
+
+	private static IColumn<CoinControlItemViewModelBase> IndicatorsColumn()
+	{
+		return new TemplateColumn<CoinControlItemViewModelBase>(
+			"",
+			new FuncDataTemplate<CoinControlItemViewModelBase>((_, _) => new IndicatorsCellView(), true),
+			GridLength.Auto,
+			new ColumnOptions<CoinControlItemViewModelBase>
+			{
+				CompareAscending = SortAscending<CoinControlItemViewModelBase, int>(GetIndicatorPriority),
+				CompareDescending = SortDescending<CoinControlItemViewModelBase, int>(GetIndicatorPriority)
+			});
+	}
+
+	private static IColumn<CoinControlItemViewModelBase> PrivacyScore()
```

Comment:
The name of this should be `PrivacyColumn()`. Or rather `AnonymityScoreColumn()` which is more telling.